src/build.c

```diff
@@ -977,26 +964,19 @@ static GPid build_run_cmd(GeanyDocument *doc, guint cmdindex)
 
 		if (vc->skip_run_script)
 		{
-			SETPTR(vte_cmd_nonscript, utils_get_utf8_from_locale(vte_cmd_nonscript));
-			vte_cmd = g_strconcat(vte_cmd_nonscript, "\n", NULL);
-			g_free(vte_cmd_nonscript);
+			SETPTR(run_cmd, utils_get_utf8_from_locale(run_cmd));
+			vte_cmd = g_strconcat(run_cmd, "\n", NULL);
 		}
 		else
-			vte_cmd = g_strconcat("\n/bin/sh ", RUN_SCRIPT_CMD, "\n", NULL);
+			vte_cmd = g_strconcat("\n/bin/sh ", run_cmd, "\n", NULL);
 
-		/* change into current directory if it is not done by default */
```

Comment:
Wrong. We have to change always since the document's directory may be different from the run directory set in the "set build commands" dialog.